libs/vertexai/langchain_google_vertexai/_anthropic_utils.py

```diff
@@ -59,14 +59,42 @@ def _format_image(image_url: str) -> Dict:
     }
 
 
-def _format_message_anthropic(message: Union[HumanMessage, AIMessage]):
-    role = _message_type_lookups[message.type]
+def _get_cache_control(message: BaseMessage) -> Optional[Dict[str, Any]]:
+    """Extract cache control from message's additional_kwargs or content block."""
+    return (
+        message.additional_kwargs.get("cache_control")
+        if isinstance(message.additional_kwargs, dict)
+        else None
+    )
+
+
+def _format_text_content(
+    text: str, cache_control: Optional[Dict[str, Any]] = None
+) -> Dict[str, Union[str, Dict[str, Any]]]:
+    """Format text content with optional cache control."""
+    content: Dict[str, Union[str, Dict[str, Any]]] = {"type": "text", "text": text}
+    if cache_control:
+        content["cache_control"] = cache_control
+    return content
+
+
+def _format_message_anthropic(message: Union[HumanMessage, AIMessage, SystemMessage]):
+    """Format a message for Anthropic API.
+
+    Args:
+        message: The message to format. Can be HumanMessage, AIMessage, or SystemMessage.
+
+    Returns:
+        A dictionary with the formatted message, or None if the message is empty.
+    """  # noqa: E501
     content: List[Dict[str, Any]] = []
 
     if isinstance(message.content, str):
         if not message.content.strip():
             return None
-        content.append({"type": "text", "text": message.content})
+        content.append(
+            _format_text_content(message.content, _get_cache_control(message))
```

Comment:
nits: maybe it would be more readable if _format_text_content expects a message only, and handles cache control itself, wdyt?

Reply:
@lkuligin 
Hey, thank you all for the review!
That’s a good observation. 
To avoid duplicating logic and keep the code more readable, I keep the code in the same if statement.